src/test/java/net/snowflake/client/pooling/LogicalConnectionLatestIT.java

```diff
@@ -28,16 +27,17 @@ public void testLogicalConnection() throws SQLException {
     poolDataSource = setProperties(poolDataSource);
     PooledConnection pooledConnection = poolDataSource.getPooledConnection();
     Connection logicalConnection = pooledConnection.getConnection();
-    Statement statement =
+    try (Statement statement =
         logicalConnection.createStatement(
             ResultSet.TYPE_FORWARD_ONLY,
             ResultSet.CONCUR_READ_ONLY,
-            ResultSet.CLOSE_CURSORS_AT_COMMIT);
-    ResultSet resultSet = statement.executeQuery("show parameters");
-    assertTrue(resultSet.next());
-    assertFalse(logicalConnection.isClosed());
-    assertEquals(ResultSet.CLOSE_CURSORS_AT_COMMIT, logicalConnection.getHoldability());
-    statement.close();
+            ResultSet.CLOSE_CURSORS_AT_COMMIT)) {
+      try (ResultSet resultSet = statement.executeQuery("show parameters")) {
```

Comment:
we need statement.close().

Reply:
Actually we don't - this is why I introduced try-with-resources, it is automatically closed at the end of the `try` block.